mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -358,6 +358,8 @@ pub struct NexusNvmeParams {
     pub(crate) max_cntlid: u16,
     /// NVMe reservation key for children
     pub(crate) resv_key: u64,
+    /// NVMe preempt key for children, 0 to not preempt
+    pub(crate) preempt_key: u64,
```

Comment:
Wouldn't `Option<std::num::NonZeroU64>` be cleaner here?

Reply:
I wasn't aware of this structure, I'll look into refactoring to use this.

Reply:
Tbh, I've never actually used any of those `NonZero` types yet. My main suggestion was about using `Option` but seeing that you've no use for the `0` value here, I suggested `Option<NonZeroU64>` instead of `Option<u64>`.